### PR TITLE
Fixed `std::initializer_list` assignment issues for open-addressing containers

### LINKED ISSUE #276: Missing assignment operator=(std::initializer_list<value_type>)
The boost.unordered containers are missing the assignment operator taking an `std::initializer_list` argument. While it may seem redundant, this can be needed in a few special cases - in particular when using custom allocators which lack default initialization (as many stateful allocators are)

### PR DESCRIPTION
**Pull request by joaquintides**
Fixes #276.

### CODE REVIEW

File: include/boost/unordered/unordered_flat_map.hpp [hunk 1]
```diff
@@ -202,6 +203,13 @@ namespace boost {
         return *this;
       }
 
+      unordered_flat_map& operator=(std::initializer_list<value_type> il)
+      {
+        clear();
+        insert(il);
```
Comment (k3DW) [inline include/boost/unordered/unordered_flat_map.hpp:209]: You could save a function call by instead writing
```cpp
insert(il.begin(), il.end());
```

**Comment by pdimov** [comment]
That's fine in principle but I prefer writing tests that reflect actual use. Nobody's going to assign `std::initializer_list<value_type>()` to a container; more probable user code would be to assign something like `{{k1, v1}, {k2, v2}}`.

File: include/boost/unordered/unordered_flat_map.hpp (hunk 1 above)
Comment (joaquintides) [reply include/boost/unordered/unordered_flat_map.hpp]: Done, thank you!

**Comment by joaquintides** [comment]
> That's fine in principle but I prefer writing tests that reflect actual use. Nobody's going to assign std::initializer_list<value_type>() to a container; more probable user code would be to assign something like {{k1, v1}, {k2, v2}}.

Fixed [here](https://github.com/boostorg/unordered/pull/277/commits/cc8b40345ef78ea191c4e6976e0f8d0b4365131f).

**Pull request merged**